code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,354 @@
+						addtimer(CALLBACK(src, .proc/spawn_elite, elitemind), 100)
+						return
+					else
+						visible_message("<span class='boldwarning'>The stirring stops, and nothing emerges.  Perhaps try again later.</span>")
+						activity = TUMOR_INACTIVE
+						activator = null
+						return
+				else
+					addtimer(CALLBACK(src, .proc/spawn_elite), 30)
+				return
+		if(TUMOR_PASSIVE)
+			if(istype(user, /mob/living/carbon/human))
+				activity = TUMOR_ACTIVE
+				visible_message("<span class='boldwarning'>The tumor convulses as your arm enters its radius.  Your instincts tell you to step back.</span>")
+				activator = user
+				INVOKE_ASYNC(src, .proc/arena_trap)
+				if(boosted)
+					SEND_SOUND(mychild, sound('sound/effects/magic.ogg'))
+					to_chat(mychild, "<b>Someone has activated your tumor.  You will be returned to fight shortly, get ready!</b>")
+				addtimer(CALLBACK(src, .proc/return_elite), 30)
+				return
+
+				
+obj/structure/elite_tumor/proc/spawn_elite(var/mob/dead/observer/elitemind)
+	var/selectedspawn = pick(potentialspawns)
+	mychild = new selectedspawn(loc)
+	mychild.myparent = src
+	visible_message("<span class='boldwarning'>[mychild] emerges from the tumor!</span>")
+	playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
+	if(boosted)
+		mychild.key = elitemind.key
+		mychild.sentience_act()
+	INVOKE_ASYNC(src, .proc/arena_trap)
+	return
+
+obj/structure/elite_tumor/proc/return_elite()
+	mychild.forceMove(loc)
+	visible_message("<span class='boldwarning'>[mychild] emerges from the tumor!</span>")
+	playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
+	mychild.revive(full_heal = TRUE, admin_revive = TRUE)
+	if(boosted)
+		mychild.maxHealth = mychild.maxHealth * 2
+		mychild.health = mychild.maxHealth
+	return
+		
+/obj/structure/elite_tumor/Initialize()
+	. = ..()
+	AddComponent(/datum/component/gps, "Menacing Signal")
+		
+/obj/structure/elite_tumor/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/organ/regenerative_core) && activity == TUMOR_INACTIVE && !boosted)
+		var/obj/item/organ/regenerative_core/core = W
+		if(core.preserved)
+			visible_message("<span class='boldwarning'>As [user] drops the core into the tumor, the tumor appears to swell.</span>")
+			icon_state = "advanced_tumor"
+			boosted = TRUE
+			light_range = 6
+			desc = "[desc]  This one seems to glow with a strong intensity."
+			qdel(core)
+			return TRUE
+			
+/obj/structure/elite_tumor/proc/arena_trap()
+	var/turf/T = get_turf(src)
+	if(T && activity == TUMOR_ACTIVE)
+		for(var/t in RANGE_TURFS(12, T))
+			if(t && get_dist(t, T) == 12)
+				var/obj/effect/temp_visual/elite_tumor_wall/newwall
+				newwall = new /obj/effect/temp_visual/elite_tumor_wall(t, src)
+				newwall.activator = src.activator
+				newwall.ourelite = src.mychild
+		addtimer(CALLBACK(src, .proc/arena_checks), 50)
+		return
+
+/obj/structure/elite_tumor/proc/arena_checks()
+	if(src) //Checking to see if we still exist
```

Comment:
QDELETED check is necessary since without it, if the elite or activator falls into a chasm on lavaland, things will break.  The ones in border_check make sure we don't play the teleportation noise and sound when one of them has been QDELETED by the chasm, and the ones in fighters_check make sure the fight ends if someone falls into a chasm.